source/guide_synapse.rst

```diff
@@ -0,0 +1,412 @@
+.. author:: 927589452
+.. highlight:: console
+
+.. tag:: python
+
+.. sidebar:: Logo
+
+  .. image:: _static/images/matrix-logo.svg
+      :align: center
+
+######
+Synapse
+######
+
+.. tag_list::
+
+Synapse_ is the reference implementation of a matrix server.
+Matrix_ is federated chat protocol aiming to replace xmpp.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * :lab:`Postgresql <guide_postgresql>`
+  * :manual:`supervisord <daemons-supervisord>`
+
+License
+=======
+
+All relevant legal information can be found here
+
+  * https://github.com/matrix-org/synapse/blob/master/LICENSE
+
+Prerequisites
+=============
+
+You need a running :lab:`Postgresql <guide_postgresql>` database server, a dedicated user with a secure password and database for synape.
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ createuser synapse -P
+  Enter password for new role:
+  Enter it again:
+  [isabell@stardust ~]$ createdb \
+    --encoding=UTF8 \
+    --owner="synapse" \
+    --template=template0 \
+    synapse
+  [isabell@stardust ~]$
+
+Installation
+============
+
+We will install synapse using pip, which makes it quite easy,
+but we start with creating a virtual environment for synapse to run in.
+
+.. code-block:: console
+  :emphasize-lines: 1,2,8,18,27,37
+
+  [isabell@stardust ~]$ mkdir -p ~/synapse
+  [isabell@stardust ~]$ virtualenv -p python3.6 ~/synapse/env
+    Running virtualenv with interpreter /usr/bin/python3.6
+    Using base prefix '/usr'
+    New python executable in /home/matrites/synapse/env/bin/python3.6
+    Also creating executable in /home/matrites/synapse/env/bin/python
+    Installing setuptools, pip, wheel...done.
+  [isabell@stardust ~]$ source ~/synapse/env/bin/activate
+  (env) [isabell@stardust ~]$ pip install --upgrade pip
```

Comment:
The installation works fine using both the default pip and setuptools. Please skip these steps to make the guide a bit shorter.